README.md

```diff
@@ -1,35 +1,55 @@
+### Run code against a Deephaven server
 
-![Connect to Community Server](./docs/assets/dhc-connect-to-server.gif)
+You can execute files as queries against a running Deephaven server with this extension. To do so, your file needs to match the language of the server you are connecting to (Python or Groovy). Atop the file, click the `Run Deephaven File` action.
 
-A new connection will show up in the "CONNECTIONS" tree, and the "PANELS" should show any variables exposed on the connection. To disconnect, hover over the connection item and click the trash icon.
+![Run Deephaven file](./docs/assets/dhc-connect-to-server.gif)
+
+A new connection will appear in the [`CONNECTIONS` panel](#connections) on the left hand side of VS Code when you execute the code. Additionally, the [`PANELS` tree](#panels) below will show any variables exposed in the connection. To disconnect, hover over the connection item and click the trash icon.
 
 ### Markdown Code Blocks
-Python and Groovy codeblocks in Markdown files can be run by clicking the "Run Deephaven Block" action directly above the codeblock.
 
-![Markdown Codeblocks](docs/assets/markdown-codeblocks.png)
+The extension also supports running code blocks in markdown files. Like with [running entire files](#run-code-against-a-deephaven-server), the code block must match the language of the server you are connecting to. Atop the code block, click the `Run Deephaven Block` action.
+
+![Run Deephaven block](docs/assets/markdown-codeblocks.png)
 
 ## Configuration
 
-A single Community server `http://localhost:10000/` is configured by default and doesn't require any additional config. Additional connections can be configured in `VS Code` settings.
+By default, the extension is configured to connect to a single Community server hosted at `http://localhost:10000`.
+
+To add more servers or change the configuration of the default server, head to VS Code settings. This opens a settings UI, where you can search for the extension with `@ext:deephaven.vscode-deephaven`. It shows two items you can configure:
 
-> Note: There are some `VS Code` bugs that cause some flakiness with the `run` button. `VS Code` `v90` introduced an optional `workbench.editor.alwaysShowEditorActions` setting. Setting this to `true` improves the experience here. Namely the run button will not disappear when running commands or selecting its dropdown. See https://github.com/deephaven/vscode-deephaven/issues/1 for more details.
+![img](docs/assets/dh-settings.png)
+
+> Note: There are some known VS Code bugs that can cause flakiness with the `run` button. VS Code v90 introduced an optional setting `workbench.editor.alwaysShowEditorActions`. Set this to `true` both improves the experience and prevents the run button from disappearing when running commands or selecting the dropdown. See [here](https://github.com/deephaven/vscode-deephaven/issues/1) for more details.
```

Comment:
Grammar quibble. Not sure `Set this` matches the tense of `improves and prevents`